### ISSUE TITLE
Turn off max expression depth limit

### ISSUE THREAD

**Issue by zachary62** [duckdb/duckdb#3549]
I'm issuing a large query with >1000 cases.
I got the following errors:
Parser Error: Max expression depth limit of 1000 exceeded

Is there any way to turn off the limit, or increase it? Thank you!

**Comment by Alex-Monahan** [comment]
Hello!

Could you please provide the query and your environment details (OS, DuckDB client and version)? It will help us debug as well as potentially make a test to prevent future regressions. Thanks!

**Comment by zachary62** [comment]
My query is very large with >1000 CASE WHEN END
I think it's not a bug, but some special query that touches the limit.

Environment:
    OS: Linux
    DuckDB Version: 0.3.3
    DuckDB Client: python

**Comment by Alex-Monahan** [comment]
One other thought for a workaround! Could you create a mapping table and use a join instead of those case when statements?

**Comment by zachary62** [comment]
How?
Let say I have a table R(A,B,C)
I have 
CASE1: A=1
CASE2: A=2 AND C =3
CASE3: A=3 OR B=2
CASE4: A != 2
...
It would be tricky to express OR, != ... predicates right?

**Comment by hannes** [comment]
The expression depth limit is there to avoid stack overflow. This limit is defined in `transformer.hpp` and you are of course welcome to change it yourself and recompile.

**Comment by zachary62** [comment]
Thank you for the suggestion! I temporarily solve this error by create temporal tables to store intermediate results.

**Comment by Mytherin** [comment]
Large case expressions shouldn't trigger the expression depth limit, see [e.g. our big_case test](https://github.com/duckdb/duckdb/blob/master/test/sql/function/generic/big_case.test#L9). Could you share the exact query that triggers this?

I agree that we should make the expression depth limit configurable though.

**Comment by zachary62** [comment]
Hi Mark, thanks for the change.
For the "large case expression", let me rephrase my issue. I have a query with >1000 "WHEN CASE END", not one When but >1000 cases. According to my experiments, >1000 WHEN triggers the limit.

**Comment by adityashakya** [comment]
I get this when I query 1000 parquet by using "UNION ALL" expression for each file as "select * from read_parquet(file_path_n)". Any suggestions here?

error message:
ParserException: Parser Error: Max expression depth limit of 1000 exceeded. Use "SET max_expression_depth TO x" to increase the maximum expression depth.

**Comment by Mytherin** [comment]
You can use a list in `read_parquet`, e.g. `select * from read_parquet(['file1.parquet', 'file2.parquet', ...])`

**Comment by adityashakya** [comment]
thanks, that worked perfectly!

**Comment by grampusluan** [comment]
I also had this problem with the following code:
`a = dk.read_parquet(fnLst)
sql = '''
        select Contract,Date,SortTime,Session,Volume,BV1,SV1,
        Volume-lag(Volume,1,0) over (partition by Contract,Date order by SortTime) as vol
        from a
'''
res = dk.sql(sql)
res.limit(5).show()

roll = 'partition by Contract,Date order by SortTime rows between 1 following and 300 following'
sql = '''
        select Contract,Date,Volume,
        avg(BV1) over ({}) as bv1,
        avg(SV1) over ({}) as sv1,
        avg(vol) over ({}) as v
        from res where Session='A'
'''.format(roll,roll,roll)
res = dk.sql(sql) # this line works fine
res.limit(5).show() # this line got Max expression depth error! `

any suggestions?

**Comment by MLpranav** [comment]
Hi, I'm getting `Parser Error: Maximum tree depth of 1000 exceeded in logical planner` on the following queries:

```
ATTACH 'old.db' AS db1;
ATTACH 'new.db' AS db2;
COPY FROM DATABASE db1 TO db2;
```

```
EXPORT DATABASE 'backup';
```

I am trying to re-create (similar to sqlite VACUUM) a database (1.5 GB, 2000 tables will 1-2k entries each).

I am initially creating the database in Python and it is taking 25% more space if I insert the data periodically (at every 5% progress) instead of all at once in the end (which is not suitable for memory consumption reasons).

Any suggestions will be appreciated. Thanks.

**Comment by Mytherin** [comment]
> Hi, I'm getting `Parser Error: Maximum tree depth of 1000 exceeded in logical planner` on the following queries:
> 
> ```
> ATTACH 'old.db' AS db1;
> ATTACH 'new.db' AS db2;
> COPY FROM DATABASE db1 TO db2;
> ```
> 
>  ```
> EXPORT DATABASE 'backup';
> ```
> 
> I am trying to re-create (similar to sqlite VACUUM) a database (1.5 GB, 2000 tables will 1-2k entries each).
> 
> I am initially creating the database in Python and it is taking 25% more space if I insert the data periodically (at every 5% progress) instead of all at once in the end (which is not suitable for memory consumption reasons).
> 
> Any suggestions will be appreciated. Thanks.
> 
> @Mytherin

I found this myself recently - I think it should be fixed by https://github.com/duckdb/duckdb/pull/13889. Could you give that a shot?
